File: nodes/array-split/array-split.process.py
async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  mode = settings.get("mode", "size")
  size = int(settings.get("size", 2))
  
  # Handle count with special care for 0
  count_setting = settings.get("count")
  if count_setting is None:
    count = 2
  else:
    count = int(count_setting)
  
  chunks = []
  
  if mode == "size":
    # Split into chunks of specified size
    for i in range(0, len(array), size):
      chunks.append(array[i:i + size])
  
  elif mode == "count":
    # Split into specified number of chunks
    if count <= 0:
      chunks = [array[:]]  # Return the entire array as a single chunk
    else:
      chunk_size = (len(array) + count - 1) // count  # Ceiling division
      for i in range(0, len(array), chunk_size):
        chunks.append(array[i:i + chunk_size])
  
  return {
    "chunks": chunks
  } 

File: sdks/nodejs/nodes/array-split/array-split.process.js
export default async ({inputs, settings, config, nodeConfig}) => {
  const array = inputs.array || [];
  const mode = settings.mode || 'size';
  const size = parseInt(settings.size) || 2;
  let count = parseInt(settings.count);
  
  // Default count to 2 only if it's not explicitly set to 0
  if (isNaN(count)) {
    count = 2;
  }
  
  let chunks = [];
  
  if (mode === 'size') {
    // Split into chunks of specified size
    for (let i = 0; i < array.length; i += size) {
      chunks.push(array.slice(i, i + size));
    }
  } else if (mode === 'count') {
    // Split into specified number of chunks
    if (count <= 0) {
      chunks = [array.slice()]; // Return the entire array as a single chunk
    } else {
      const chunkSize = Math.ceil(array.length / count);
      for (let i = 0; i < array.length; i += chunkSize) {
        chunks.push(array.slice(i, i + chunkSize));
      }
    }
  }
  
  return { chunks };
}; 